apps/frontend/src/libs/components/input/input.tsx

```diff
@@ -5,35 +5,76 @@ import {
 	type FieldValues,
 } from "react-hook-form";
 
-import { useFormController } from "~/libs/hooks/hooks.js";
+import { useCallback, useFormController } from "~/libs/hooks/hooks.js";
 
 type Properties<T extends FieldValues> = {
+	className: string;
+	clickIcon?: (event_: React.BaseSyntheticEvent) => void;
 	control: Control<T, null>;
 	errors: FieldErrors<T>;
+	hasIcon?: string;
+	iconSrc?: string;
+	inputChange?: (event_: React.BaseSyntheticEvent) => void;
 	label: string;
 	name: FieldPath<T>;
 	placeholder?: string;
-	type?: "email" | "text";
+	required?: string;
+	type?: "email" | "password" | "text";
 };
 
 const Input = <T extends FieldValues>({
+	className,
+	clickIcon,
 	control,
 	errors,
+	hasIcon,
+	iconSrc,
+	inputChange,
 	label,
 	name,
 	placeholder = "",
+	required,
 	type = "text",
 }: Properties<T>): JSX.Element => {
 	const { field } = useFormController({ control, name });
 
 	const error = errors[name]?.message;
 	const hasError = Boolean(error);
 
+	const handleChange = useCallback(
+		(event: React.BaseSyntheticEvent) => {
+			field.onChange(event);
+			if (inputChange) {
+				inputChange(event);
+			}
+		},
+		[field, inputChange],
+	);
+
 	return (
-		<label>
-			<span>{label}</span>
-			<input {...field} placeholder={placeholder} type={type} />
-			{hasError && <span>{error as string}</span>}
+		<label className={className}>
+			<span className="label">{label}</span>
```

Comment:
Consider using the same prop for label and placeholder, this will reduce the amount of props to pass to this component, therefore making the code back at sign-up-form.tsx component more readable

Reply:
but, i see emails have different placeholder. So, i can't combine "labels" and "placeholders" into one unit